utils/funcexp/func_mod.cpp

```diff
@@ -237,12 +237,12 @@ double Func_mod::getDoubleVal(Row& row,
                 }
                 else
                 {
-                    int128_t scaleDivisor;
-                    datatypes::getScaleDivisor(scaleDivisor, d.scale);
-                    int128_t value = d.s128Value / scaleDivisor;
-                    int128_t lefto = d.s128Value % scaleDivisor;
-                    datatypes::TFloat128 tmp((__float128) (value % div) + (__float128) lefto / scaleDivisor);
-                    mod = static_cast<double>(tmp);
+                    auto intAndFract = d.getIntegralAndFractional<datatypes::TFloat128>();
+
+                    datatypes::TSInt128 integralRemainder = intAndFract.first % div;
+                    mod = static_cast<double>(integralRemainder.toTFloat128() +
+                                                   intAndFract.second);
+
```

Comment:
I suggest to move this code to a new private/protected method in Func_mod. Say:

`IDB_Decimal doDecimal();`

So here you can just do:

`return static_cast<double>(doDecimal())`

Reply:
Good point. I will deliver this outside the scope of this patch.

Reply:
Done.